engine/examples/server/main.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"io"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/googollee/go-socket.io/engine"
+)
+
+func main() {
+	eio, err := engine.New(
+		engine.OptionTransports("polling", "sse", "websocket"),
+		engine.OptionJSONP(4), // polling with jsonp `__eio[4]("packet data")`
+		engine.OptionPingInterval(25*time.Second),
+		engine.OptionPingTimeout(50*time.Second),
+		engine.OptionMaxBufferSize(1*1024*1024), // 1 MiB
+		engine.OptionLogLevel(engine.LogDebug),
+		engine.OptionLogger(nil),
+	)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	eio.With(func(ctx *engine.Context) {
```

Comment:
`With()` adds a middleware to the packet processing. `ctx.Next()` continues the processing. All data from the client are in the context struct now, including Request.

As users could add a logging middleware, I tend to remove `OptionLogger()` and `OptionLogLevel()`.